src/filehash.rs

```diff
@@ -71,10 +75,11 @@ impl Filehash {
 
     /// Checks whether the calculated hash of `file_str` is equal to the hash stored
     /// in the file `hash_file_str`
-    pub fn check_hash_currency(hash_file_str: &str, file_str: &str) -> Result<String> {
+    pub fn check_hash_currency<'a, P: Into<&'a PathBuf>>(hash_file: P,
```

Comment:
You may consider using an unreferenced PathBuf to avoid desugaring by setting a lifetime like this: 

`pub fn check_hash_currency<P: Into<PathBuf>>(hash_file: P,
                                                    file_str: &str) -> Result<String>`